sdk/src/assertions/actions.rs

```diff
@@ -274,6 +275,40 @@ impl Action {
         }
     }
 
+    /// Returns an array of the [`hashed_uri::HashedUri`]s that are ingredients for this action.
+    pub fn ingredients(&self) -> Result<Vec<hashed_uri::HashedUri>> {
+        let mut result = Vec::new();
+        let value = self.get_parameter("ingredients");
+        if let Some(Value::Array(values)) = value {
+            for value in values {
+                let hashed_uri: Result<hashed_uri::HashedUri> = match value {
+                    Value::Map(_) | Value::Tag(_, _) => serde_cbor::to_vec(&value)
+                        .and_then(|v| serde_cbor::from_slice::<hashed_uri::HashedUri>(&v))
+                        .map_err(|e| {
+                            crate::Error::AssertionSpecificError(format!(
+                                "Failed to parse Action ingredient: {e}"
+                            ))
+                        }),
+                    Value::Bytes(bytes) => serde_cbor::from_slice::<hashed_uri::HashedUri>(bytes)
+                        .map_err(|e| {
+                            crate::Error::AssertionSpecificError(format!(
+                                "Failed to parse Action ingredient: {e}"
+                            ))
+                        }),
+                    _ => Err(crate::Error::AssertionSpecificError(
+                        "Unsupported CBOR type for HashedUri".to_string(),
+                    )),
+                };
+                result.push(hashed_uri?);
+            }
+            Ok(result)
+        } else {
+            Err(crate::Error::AssertionSpecificError(
+                "Unsupported CBOR type for HashedUri".to_string(),
+            ))
+        }
+    }
+
```

Comment:
I experimented with defining the `ParametersMap` with serde here: https://github.com/contentauth/c2pa-rs/compare/main...ok-nick/parameters-action

I think that would be most ideal scenario so we don't have to deal with manually decoding it like here.

Reply:
Please submit that as a PR. That is a MUCH better way to do all of this and long overdue. Thank you!

Reply:
We can update this based on that PR

Reply:
In the mean time I just removed this method